src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2327,6 +2331,229 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
+                    b.Value.Delete();   // not in new list so delete it
+                }
+            }
+
+            foreach (DataBreakpoint b in breakpoints)
+            {
+                if (m_dataBreakpoints.ContainsKey(b.DataId))
+                {   // already created
+                    IDebugBreakpointRequest2 breakpointRequest;
+                    if (m_dataBreakpoints[b.DataId].GetBreakpointRequest(out breakpointRequest) == 0 &&
+                                breakpointRequest is AD7BreakPointRequest ad7BPRequest)
+                    {
+                        // Check to see if this breakpoint has a condition that has changed.
+                        if (!StringComparer.Ordinal.Equals(ad7BPRequest.Condition, b.Condition))
+                        {
+                            // Condition has been modified. Delete breakpoint so it will be recreated with the updated condition.
+                            var toRemove = m_dataBreakpoints[b.DataId];
+                            toRemove.Delete();
+                            m_dataBreakpoints.Remove(b.DataId);
+                        }
+                        else
+                        {
+                            if (ad7BPRequest.BindResult != null)
+                            {
+                                response.Breakpoints.Add(ad7BPRequest.BindResult);
+                            }
+                            else
+                            {
+                                response.Breakpoints.Add(new Breakpoint()
+                                {
+                                    Id = (int)ad7BPRequest.Id,
+                                    Verified = true,
+                                    Line = 0
+                                });
+
+                            }
+                            continue;
+                        }
+                    }
+                }
+
+                // Bind the new data bp
+                if (!m_dataBreakpoints.ContainsKey(b.DataId))
+                {
+                    int hr = HRConstants.S_OK;
+
+                    string[] splits = b.DataId.Split(',');
+                    string address = splits[0];
+                    uint size = uint.Parse(splits[1], CultureInfo.InvariantCulture);
+
+                    AD7BreakPointRequest pBPRequest = new AD7BreakPointRequest(address, size);
+                    hr = m_engine.CreatePendingBreakpoint(pBPRequest, out IDebugPendingBreakpoint2 pendingBp);
+
+                    if (hr == HRConstants.S_OK && pendingBp != null)
+                    {
+                        hr = pendingBp.Bind();
+                    }
+
+                    if (hr == HRConstants.S_OK)
+                    {
+                        newBreakpoints[b.DataId] = pendingBp;
+                        response.Breakpoints.Add(new Breakpoint()
+                        {
+                            Id = (int)pBPRequest.Id,
+                            Verified = true,
```

Comment:
>Verified = true, [](http://example.com/codeflow?start=28&length=16)

<Aside>I am fairly certain this wrong, but looks like we have the same bug throughout OpenDebugAD7 -- a breakpoint isn't verified until we get a bound breakpoint event from the engine.</Aside>